### PR TITLE
feat: Download dem file from JAXA if not provided by the user

### PR DESCRIPTION
**Pull request by aryalsaurav**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation
- [ ] 🧑‍💻 Refactor
- [ ] ✅ Test
- [ ] 🤖 Build or CI
- [ ] ❓ Other (please specify)

## Describe this PR

This PR introduces a feature where if the Project creator wants DEM file to be automatically uploaded instead of him uploading the file. Scrapy Bot will crawl the "Jaxa" website for the dem file and if there are multiple images that falls in that selected range then it merges all into one DEM file and later uploaded to S3 bucket.

## Review Guide

Notes for the reviewer. How to test this change?

## Checklist before requesting a review

- 📖 Read the HOT Code of Conduct: <https://docs.hotosm.org/code-of-conduct>
- 👷‍♀️ Create small PRs. In most cases, this will be possible.
- ✅ Provide tests for your changes.
- 📝 Use descriptive commit messages.
- 📗 Update any related documentation and include any relevant screenshots.

### CODE REVIEW

File: src/backend/app/jaxa/jaxa/settings.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Scrapy settings for jaxa project
+#
```
Comment (Pradip-p) [inline src/backend/app/jaxa/jaxa/settings.py]: I think, you set up a Scrapy app(Jaxa), which is great! However, I highly recommend using Scrapy's **Core API**. It simplifies the workflow and is perfect for single-page scripts without requiring a full project setup.  

Here’s an example:  
```python
from scrapy.crawler import CrawlerRunner
from scrapy import Spider
from twisted.internet import reactor

class ExampleSpider(Spider):
    name = "example"
    start_urls = ["https://example.com"]

    def parse(self, response):
        print("Scraped content:", response.body[:100])

runner = CrawlerRunner()
runner.crawl(ExampleSpider)
runner.join().addBoth(lambda _: reactor.stop())
reactor.run()
```

Comment (Pradip-p) [inline src/backend/app/jaxa/jaxa/settings.py:2]: I think, you set up a Scrapy app(Jaxa), which is great! However, I highly recommend using Scrapy's **Core API**. It simplifies the workflow and is perfect for single-page scripts without requiring a full project setup.  

Here’s an example:  
```python
from scrapy.crawler import CrawlerRunner
from scrapy import Spider
from twisted.internet import reactor

class ExampleSpider(Spider):
    name = "example"
    start_urls = ["https://example.com"]

    def parse(self, response):
        print("Scraped content:", response.body[:100])

runner = CrawlerRunner()
runner.crawl(ExampleSpider)
runner.join().addBoth(lambda _: reactor.stop())
reactor.run()
```

File: src/backend/app/jaxa/tif_spider.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        self.coordinates = coordinates.split(",")
+
+    headers = {
+        "authority": "www.eorc.jaxa.jp",
+        "path": "/ALOS/en/aw3d30/data/html_v2404/xml/{caption}_5_5.xml",
+        "method": "GET",
+        "accept": "application/xml, text/xml, */*; q=0.01",
+        "accept-encoding": "gzip, deflate, br, zstd",
+        "accept-language": "en-US,en;q=0.9",
+        "authorization": "Basic cHJhZGlwdGhhcGEubmF4YUBnbWFpbC5jb206YXczZDMw",
```
Comment (nrjadkry) [inline src/backend/app/jaxa/tif_spider.py]: Is this the authorization token you have hardcoded ?

**Comment by bijayrauniyar0** [comment]
- Add radio buttons to ask user if they want to upload DEM or they need from server

### Screenshots
image

**Comment by spwoodcock** [comment]
> - Add radio buttons to ask user if they want to upload DEM or they need from server
> 
> ### Screenshots
> image
> 

Nice 👍

Is the text supposed to say 'approval for lock task'?

Also having it marked mandatory doesn't make sense if you provide a default and its a radio.

Perhaps it might help adding a small bit of explanation:

'We provide a global elevation model that should be suitable for most cases, but you may wish to upload a more detailed custom model in certain scenarios (for example steep mountainous terrain)'

**Review by subashtiwari1010 — APPROVED** [review]

**Review by nrjadkry — APPROVED** [review]

**Comment by smathermather** [comment]
Brilliant!

**Pull request merged**